Extension/src/LanguageServer/client.ts

```diff
@@ -620,6 +627,16 @@ class DefaultClient implements Client {
         this.model.tagParserStatus.Value = notificationBody.status;
     }
 
+    private updateInactiveRegions(params: InactiveRegionParams): void {
+        let renderOptions: vscode.DecorationRenderOptions = {
+            color: "rgba(255,0,255,0.5)"
+        };
+        let textEditorDecoration = vscode.window.createTextEditorDecorationType(renderOptions);
+        
+        let editor = vscode.window.visibleTextEditors.find(e => e.document.uri.toString() == params.uri); // TODO change to document.uri == params.uri
+        editor.setDecorations(textEditorDecoration, params.ranges);
```

Comment:
>editor.setDecorations [](start = 8, length = 21)

with each notification, I think we'll need to save the ranges that come in.  This way we can clear out the old decorations before setting the new ones.

Reply:
It looks like you might also be able to call dispose on the TextEditorDecorationType to remove all decorations associated with that object.  If that's the case, then my previous comment about creating the TextEditorDecorationType only once would be invalid.  You might be able to just keep a reference to the previous TextEditorDecorationType and dispose it before creating a new one and assigning it to the new ranges.

---
In reply to: [163304083](https://github.com/Microsoft/vscode-cpptools/pull/1482#discussion_r163304083) [](ancestors = 163304083)

Reply:
That sounds like a fairly tidy way to manage the decoration types. I will make sure to investigate that possibility first!